week2/index.js

```diff
@@ -0,0 +1,163 @@
+"use-strict";
+
+const fs = require('fs');
+const TODO_FILE = process.env['TODO_file'] || 'todo.json'
+const HELP_MENU = 'help.txt';
+
+let readFiles = (filename) => {
+    return new Promise((resolve, reject) => {
+        fs.readFile(filename, "utf8", (err, data) => {
+            if (err) {
+                if (err.code === "ENOENT") {
+                    // No such file or directory
+                    console.log("Sorry, this file doesn't exist.");
+                } else if (err.code === "EACCES") {
+                    // the user don't have a permission
+                    console.log("Sorry, but you don't have permission.");
+                } else {
+                    // any other Error
+                    console.log(` unknown error occurred: ${err.message}`);
+                }
+                process.exit();
+            }
+            resolve(data);
+
+        })
+    })
+
+};
+
+let writeFiles = (filename, dataAsString) => {
+    return new Promise((resolve, reject) => {
+        fs.writeFile(TODO_FILE, dataAsString, (err) => {
+            if (err) {
+                if (err.code === "ENOENT") {
+                    // No such file or directory
+                    console.log("Sorry, this file doesn't exist.");
+                } else if (err.code === "EACCES") {
+                    // the user don't have a permission
+                    console.log("Sorry, but you don't have permission.");
+                } else {
+                    // any other Error
+                    console.log(` unknown error occurred: ${err.message}`);
+                }
+                process.exit();
+            }
+            resolve();
+        })
+    })
+}
+
+let addTodoItem = (filename, todoList, itemText) => {
+    todoList.push({
+        TASK_NUMBER: todoList.length + 1,
+        TODO_TASK: itemText
+    });
+    writeFiles(filename, JSON.stringify(myTodoList)).then(() => console.log("\nDone Adding the task on your TODO list.\n\n")).then(() => askingLoop());

```

Comment:
This one line is too long, it should be split up like: 
```
writeFiles(filename, JSON.stringify(myTodoList)).then(() => {
  console.log("\nDone Adding the task on your TODO list.\n\n")
}).then(() => askingLoop());
```

Or, to keep all the `.then`s in line:
```
writeFiles(filename, JSON.stringify(myTodoList))
  .then(() => console.log("\nDone Adding the task on your TODO list.\n\n"))
  .then(() => askingLoop());
```